features/FEATURE_BLE/ble/gap/Events.h

```diff
@@ -63,7 +63,7 @@ struct AdvertisingReportEvent {
         advertising_sid_t SID,
         advertising_power_t txPower,
         rssi_t rssi,
-        periodic_interval_t periodicInterval,
+        uint16_t periodicInterval,
```

Comment:
Odd. Why not replace/update the constructor instead?

Reply:
As you can see, that constructor is not part of the documentation because it is not meant to be used by user code. We cannot use `friend` in this case because `friend` cannot be inherited and we don't know the vendor file that will use it. There is no reason for us to keep non usable constructor  if it is not part of the public API. 

To give a bit more context, the periodic interval returned in an advertising event can be equal to 0 if there is no periodic interval or a value comprised between [0x0006 : 0xFFFF]. The type `periodic_interval_t` is not able to represent the absence of value as it normalizes values passed in input.